File: GUI/Custom Controls/PanelTransparentTextBox.vb
Imports System.Drawing
Imports System.Windows.Forms
Imports System.Windows.Forms.VisualStyles

Partial Public Class PanelTransparentTextBox
    Inherits Panel
    Public Sub New()
        SetStyle(ControlStyles.SupportsTransparentBackColor Or ControlStyles.OptimizedDoubleBuffer Or ControlStyles.AllPaintingInWmPaint Or ControlStyles.ResizeRedraw Or ControlStyles.UserPaint, True)
        BackColor = Color.Transparent
        Dim line As New LabelDoubleBuffer With {
            .BackColor = Color.White,
            .Size = New Size(Me.Width, 2),
            .Location = New Point(0, Me.Height),
            .Dock = DockStyle.Bottom,
            .AutoSize = False
        }
        Me.Controls.Add(line)
    End Sub


End Class

File: Development/Platform/GUI/Custom Controls/Custom Controls/PanelTransparentTextBox.cs
using System.Drawing;
using System.Windows.Forms;

public partial class PanelTransparentTextBox : Panel
{
    public PanelTransparentTextBox()
    {
        SuspendLayout();
        SetStyle(ControlStyles.AllPaintingInWmPaint, true);
        SetStyle(ControlStyles.OptimizedDoubleBuffer, true);
        // ' SetStyle(ControlStyles.SupportsTransparentBackColor, True)
        SetStyle(ControlStyles.ResizeRedraw, true);
        SetStyle(ControlStyles.UserPaint, true);
        UpdateStyles();
        base.BackColor = Color.Transparent;
        var line = new LabelDoubleBuffer()
        {
            BackColor = Color.White,
            Size = new Size(Width, 2),
            Location = new Point(0, Height),
            Dock = DockStyle.Bottom,
            AutoSize = false
        };
        Controls.Add(line);
        ResumeLayout();
    }
}